src/modules/windowwalker/app/Window Walker/Components/OpenWindows.cs

```diff
@@ -94,23 +94,32 @@ public bool WindowEnumerationCallBack(IntPtr hwnd, IntPtr lParam)
         {
             Window newWindow = new Window(hwnd);
 
-            if (windows.Select(x => x.Title).Contains(newWindow.Title))
+            uint toolWindow = (uint)InteropAndHelpers.ExtendedWindowStyles.WS_EX_TOOLWINDOW;
+            uint appWindow = (uint)InteropAndHelpers.ExtendedWindowStyles.WS_EX_APPWINDOW;
+            InteropAndHelpers.GetWindowCmd owner = InteropAndHelpers.GetWindowCmd.GW_OWNER;
+            int windowLong = InteropAndHelpers.GetWindowLong(hwnd, InteropAndHelpers.GWL_EXSTYLE);

```

Comment:
Can you abstract these into Boolean properties in the `Window` class and reference the properties in the if statement below? It will make it a lot easier to follow and making your calls reusable elsewhere in this codebase.

Reply:
Done!
It is also included `IsUWPCloaked` for future use, in case there will be optional options to hide them in current desktop.

e.g.
```cs
 if (newWindow.IsWindow && newWindow.Visible && newWindow.IsOwner &&
      (!newWindow.IsToolWindow || newWindow.IsAppWindow ) && !newWindow.TaskListDeleted &&
      newWindow.ClassName != "Windows.UI.Core.CoreWindow")
      {
            switch (ListingOptions)
              {
                  case HIDE_ALL_APPS_IN_ANOTHER_DESKTOP:
                      if(!newWindow.IsWindowCloaked())
                          windows.Add(newWindow);
                      break;
                  case HIDE_ONLY_UWP_APPS_IN_ANOTHER_DESKTOP:
                      if(!newWindow.IsUWPCloaked)
                          windows.Add(newWindow);
                      break;
                  default: // lists all apps
                      windows.Add(newWindow);
                      break;
              }
       }
```